src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -878,11 +866,55 @@ public List<E> reversed() {
             return new StableReverseOrderListView<>(this);
         }
 
+        @Override
+        public List<E> subList(int fromIndex, int toIndex) {
+            int size = size();
+            subListRangeCheck(fromIndex, toIndex, size);
+            return StableSubList.fromList(this, fromIndex, toIndex);
+        }
+
         @Override
         public String toString() {
             return StableUtil.renderElements(this, "StableList", delegates);
         }
 
+        private static final class StableSubList<E> extends SubList<E> {
+
+            private StableSubList(AbstractImmutableList<E> root, int offset, int size) {
+                super(root, offset, size);
+            }
+
+            @Override
+            public List<E> reversed() {
+                return new StableReverseOrderListView<>(this);
+            }
+
+            @Override
+            public List<E> subList(int fromIndex, int toIndex) {
+                int size = size();
+                subListRangeCheck(fromIndex, toIndex, size);
```

Comment:
```suggestion
                subListRangeCheck(fromIndex, toIndex, size());
```
size is only used once, can it be simplified like this?